src/xo-app/index.js

```diff
@@ -1,54 +1,109 @@
 import React, { Component, PropTypes } from 'react'
-import { connect } from 'react-redux'
-import { Link, Route, IndexLink, IndexRoute } from 'react-router'
-import { ReduxRouter } from 'redux-router'
+import { connect, Provider } from 'react-redux'
+import store from '../store'
+import LoginForm from './login/loginForm.js'
+import VMForm from './vm/form.js'
+import { Router, Route, Link } from 'react-router'
+// from https://github.com/rackt/react-router/blob/master/upgrade-guides/v2.0.0.md#no-default-history
+import { hashHistory } from 'react-router'
 
-import About from './about'
-import Home from './home'
-import { actions } from '../store'
 
-let XoApp = class extends Component {
-  static propTypes = {
-    children: PropTypes.node,
-    counter: PropTypes.number
-  };
+const Dashboard = React.createClass({
```

Comment:
Also, a global question , where do you think the component should store its state ? 

1. use the global store : every component is a pure function but with a lot of complexity pushed up to the reducers/actions.
2. use local state only on high level component ( for example a form) , and notify store only when a change can affect the whole app state. This may lead to data duplication between the store and the  state, and top level component will be reponsible for 
3. other ? 

In 1. it's clear that all the api calls are made from the action creator. In 2, I don't know where to put them.